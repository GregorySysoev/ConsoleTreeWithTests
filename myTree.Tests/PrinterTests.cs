using System;
using Xunit;
using System.IO;
using System.Collections.Generic;
using System.Linq;
using Shouldly;

namespace myTree.Tests
{
    public class PrinterTests
    {
        public string path = "./test";
        public StringCompareShould opt = StringCompareShould.IgnoreCase;
        ListWriter ls = new ListWriter();
        Algorithm alg;

        [Fact]
        public void Print_badArgs_badArgs()
        {
            string[] args = new string[] { "-2d" };
            ls = new ListWriter();
            alg = new Algorithm(args, ls, path);

            alg.Execute();

            ls.list.ShouldMatchApproved(c => c.WithStringCompareOptions(opt));
        }

        [Fact]
        public void Print_Help_Help()
        {
            string[] args = new string[] { "--help" };
            ls = new ListWriter();
            alg = new Algorithm(args, ls, path);

            alg.Execute();

            ls.list.ShouldMatchApproved(c => c.WithStringCompareOptions(opt));
        }

        [Fact]
        public void Print_noArgs_myTreeWithoutArgs()
        {
            string[] args = new string[] { };
            ls = new ListWriter();
            alg = new Algorithm(args, ls, path);

            alg.Execute();

            ls.list.ShouldMatchApproved(c => c.WithStringCompareOptions(opt));
        }

        [Fact]
        public void Print_Size_Size()
        {
            string[] args = new string[] { "-s" };
            ls = new ListWriter();
            alg = new Algorithm(args, ls, path);

            alg.Execute();

            ls.list.ShouldMatchApproved(c => c.WithStringCompareOptions(opt));
        }

        [Fact]
        public void Print_HumanReadable_HumanReadable()
        {
            string[] args = new string[] { "-h" };
            ls = new ListWriter();
            alg = new Algorithm(args, ls, path);

            alg.Execute();

            ls.list.ShouldMatchApproved(c => c.WithStringCompareOptions(opt));
        }

        [Fact]
        public void Print_Depth1_Depth1()
        {
            string[] args = new string[] { "-d", "1" };
            ls = new ListWriter();
            alg = new Algorithm(args, ls, path);

            alg.Execute();

            ls.list.ShouldMatchApproved(c => c.WithStringCompareOptions(opt));
        }

        [Fact]
        public void Print_OrderByAlphabet_OrderByAlphabet()
        {
            string[] args = new string[] { "-o", "a" };
            ls = new ListWriter();
            alg = new Algorithm(args, ls, path);

            alg.Execute();

            ls.list.ShouldMatchApproved(c => c.WithStringCompareOptions(opt));
        }

        [Fact]
        public void Print_OrderByAlphabetRevert_OrderByAlphabetRevert()
        {
            string[] args = new string[] { "-o", "a", "-r" };
            ls = new ListWriter();
            alg = new Algorithm(args, ls, path);

            alg.Execute();

            ls.list.ShouldMatchApproved(c => c.WithStringCompareOptions(opt));
        }

        [Fact]
        public void Print_OrderBySize_OrderBySize()
        {
            string[] args = new string[] { "-o", "s", "-s" };
            ls = new ListWriter();
            alg = new Algorithm(args, ls, path);

            alg.Execute();

            ls.list.ShouldMatchApproved(c => c.WithStringCompareOptions(opt));
        }

        [Fact]
        public void Print_OrderBySizeReverse_OrderBySizeReverse()
        {
            string[] args = new string[] { "-o", "s", "-s", "-r" };
            ls = new ListWriter();
            alg = new Algorithm(args, ls, path);

            alg.Execute();

            ls.list.ShouldMatchApproved(c => c.WithStringCompareOptions(opt));
        }

        [Fact]
        public void Print_OrderByDateOfCreation_OrderByDateOfCreation()
        {
            string[] args = new string[] { "-o", "c" };
            ls = new ListWriter();
            alg = new Algorithm(args, ls, path);

            alg.Execute();

            ls.list.ShouldMatchApproved(c => c.WithStringCompareOptions(opt));
        }

        [Fact]
        public void Print_OrderByDateOfTransform_OrderByDateOfTransform()
        {
            string[] args = new string[] { "-o", "t" };
            ls = new ListWriter();
            alg = new Algorithm(args, ls, path);

            alg.Execute();

            ls.list.ShouldMatchApproved(c => c.WithStringCompareOptions(opt));
        }
    }
}
