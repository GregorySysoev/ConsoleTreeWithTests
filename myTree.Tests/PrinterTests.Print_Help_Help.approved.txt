
List of available commands:

-d --depth [num]  nesting level
-s --size  show size of files
-h --human-readable  show size of files in human-readable view {Bytes, KB, ...}
-r reverse order of elements
-o --order-by [flag] order of elements in tree. Default - by alphabet

Available flags:
s - order by size
t - order by time of last change
c - order by time of creation


Attation
If you are using 'dotnet myTree.dll' without args - you will see a whole tree
If you are using 'dotnet myTree.dll' without '--order-by [s, t, c]' - default tree will be sorted by alphabet

--help show help

Example of using:
dotnet myTree.dll -d 5 -h -o a

